### DISCUSSION TITLE
`machine` references vs identifiers

### DISCUSSION THREAD

**Discussion by dotnwat** [p-org/P#755] [Q&A] [answered]
Consider the common situation in an implementation of a system where various resource identifiers (e.g. `node_id`) are stored and cached in ways that establish relationships between the resources.

In a P model of such a system where resources are modeled as `machine`, it seems common to have to choose between generating and using identifiers, and embedding references of `machine` type to represent the same relationships.

Are there advantages / disadvantages to either approach?

**Comment by ankushdesai** [comment, marked as answer]
Just to understand a bit better; 

Is one of the reason to model resources as machine that you want to interact with the resources as well as these resources have state in them that change over time and you want to maintain that state in the resource state machines?

If yes, then I agree that modeling resource as a `stateful` state machine is the right choice and then instances of these machines represent resource instances.

**Comment by dotnwat** [comment]
>Is one of the reason to model resources as machine that you want to interact with the resources as well as these resources have state in them that change over time and you want to maintain that state in the resource state machines?

exactly. you said it much better than myself.
